crates/rslint_parser/src/ast/generated/macros.rs

```
@@ -0,0 +1,1185 @@
+//! Generated file, do not edit by hand, see `xtask/codegen`
+
+#[macro_export]
+macro_rules! map_syntax_node {
```

Would be good to have some documentation explaining how this macro is to be used. 

I don't know if we need the fallback. The only case where a `kind` has no corresponding `AstNode` is if the kind is a token kind and not a node kind.... I believe it's fine to panic in that case because all kind of weird things can happen if someone starts creating a `SyntaxNode` with a token kind (maybe a reason to separate the two enums?) 

An alternative would be to call the method `map_syntax_element` that accepts `SyntaxNode` and `SyntaxToken`. The macro then has the form `map_syntax_element(element, node_mapper, token_mapper)`

---

This is mostly a matter of leaving the choice of whether to panic or return an error to the code using the macro rather than the macro itself, but I could have the fallback branch default to `_ => unreachable!()` when its not specified